### ISSUE TITLE
Server Stuck at Processing Image

### ISSUE THREAD

**Issue by tawalaya** [mattermost/mattermost-server#20211]
Hi,
We recently had an issue with the S3 backend. After addressing the issue, we noticed that all uploads worked except images. If a user uploaded an image, the client (any) would be stuck at the processing image step. Both client and server logs showed no issue. 

Looking into the issue a bit further, I noticed that the server uploaded the image files correctly to S3, but the server is stuck processing the images. I don't know why but looking at the [code](https://github.com/mattermost/mattermost-server/blob/aa59c28b04fa45f0b02e5149ac0fbcd2b78420b9/app/file.go#L810) I was wondering why there is no timeout for upload image requests? Based on my observation, one or all image postprocessing jobs were stuck since no "_thumb" or "_preview" showed up in the S3 bucket.

I could fix the issue by rebooting the server for now.

**Comment by amyblais** [comment]
@tawalaya 
1. What device, OS and OS version do you see this on?
2. Would you be open to sharing your config settings related to files as well as this setting: https://docs.mattermost.com/configure/configuration-settings.html#allow-untrusted-internal-connections-to?
3. Are you on a supported architecture https://docs.mattermost.com/install/software-hardware-requirements.html?

**Comment by tawalaya** [comment]
Dear @amyblais , 
sorry for the delay.

1. We are running Version: 6.4.1 on bare-metal Ubuntu LTS 22.04. I checked and the latest release of mattermost also does not have timeouts for the thumbnail functions. 
2. It worked before and after rebooting everything it works again now so it's not a configuration issue. I think what happened is that during the downtime of our S3 backend, the workers used to generate thumbnails got stuck and all subsequent requests to upload an image were waiting for the stuck worker to finish. Thus, my suggestion is to add some timeout mechanism to the indicated code.
3. As far as I can tell we run unsupported architecture.

With regards to posting to the peer to peer help. I don't need help fixing the issue. I just noticed a code quality issue (missing timeout in a backend API call) while fixing the issue.

**Comment by agnivade** [comment]
@tawalaya - To confirm, you are referring to timeouts of the s3 image upload operation correct?

Also, could you clarify how did you detect that the workers were stuck at the image upload step to S3?

**Comment by tawalaya** [comment]
Yes, the issue presented itself on the client (desktop) side with a never-ending spinning loading animation when uploading any image.

Looking into the S3 bucket for a specific image that I tried to post (following the log information from the server) I noticed that the originally uploaded image made it into the bucket but the images "<name>_preview" and "<name>_thumb" did not show up. 

Looking into the `file.go` postprocessImage function I noticed that the only blocking operation in there is the waitgroup at [file.go](https://github.com/mattermost/mattermost-server/blob/aa59c28b04fa45f0b02e5149ac0fbcd2b78420b9/app/file.go#L856), thus, since neither processed image shows up in s3 and the operation continues forever all of these subroutines must be stuck somehow. 

Not knowing how the rest of the architecture deals with this I'd just add a timeout with the waitgroup to let this operation fail in these cases so that a user is not left waiting for an image to not upload. Although, I don't know if that creates issues elsewhere.
